packages/slate-editor/src/components/SearchInput/Suggestions.tsx

```diff
@@ -44,42 +46,51 @@ export function Suggestions<T>({
                 name: 'flip',
                 enabled: true,
                 options: {
-                    fallbackPlacements: ['top'],
+                    fallbackPlacements: ['top-end'],
                 },
             } satisfies Partial<FlipModifier>,
-            {
-                name: 'preventOverflow',
-                enabled: true,
-                options: {
-                    altAxis: true,
-                    mainAxis: true,
-                },
-            } satisfies Partial<PreventOverflowModifier>,
         ],
-        placement: 'bottom',
+        placement: 'bottom-end',
     });
 
     const updatePanelSizeAndPosition = useFunction(() => {
         setHeight(childrenContainer.current?.getBoundingClientRect().height);
+
+        if (container.current) {
+            const viewport = document.body.getBoundingClientRect();
+            const rect = container.current.getBoundingClientRect();
+            setMaxHeight(clamp(viewport.height - rect.top - 4, minHeight, maxHeight));
+        } else {
+            setMaxHeight(undefined);
+        }
     });
 
-    useEffect(updatePanelSizeAndPosition, [query, suggestions, maxHeight]);
+    useEffect(updatePanelSizeAndPosition, [
```

Comment:
Good improvement, but this should also happen on window resize and other events that may change available space.
Not sure if it's needed to be addressed in this PR.